lib/ansible/modules/windows/win_chocolatey_source.ps1

```diff
@@ -0,0 +1,411 @@
+    $source.$attribute = $value
+    $new_xml = Get-PrettyXml -xml $choco_config
+    if (-not $check_mode) {
+        [System.IO.File]::WriteAllText($choco_config_path, $new_xml)
+    }
+}
+
+Function New-ChocolateySource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingUserNameAndPassWordParams", "", Justification="We need to use the plaintext pass in the cmdline, also using a SecureString here doesn't make sense considering the source is not secure")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingPlainTextForPassword", "", Justification="See above")]
+    param(
+        $choco_app,
+        $name,
+        $source,
+        $source_username,
+        $source_password,
+        [switch]$encrypted_source_password,
+        $certificate,
+        $certificate_password,
+        [switch]$encrypted_certificate_password,
+        $priority,
+        $bypass_proxy,
+        $allow_self_service,
+        $admin_only
+    )
+    # build the base arguments
+    $arguments = [System.Collections.ArrayList]@($choco_app.Path,
+        "source", "add", "--name", $name, "--source", $source
+    )
+
+    # add optional arguments from user input
+    if ($null -ne $source_username) {
+        $arguments.Add("--user") > $null
+        $arguments.Add($source_username) > $null
+        $arguments.Add("--password") > $null
+        if ($encrypted_source_password) {
+            # use a placeholder for now
+            $arguments.Add("pass") > $null
+        } else {
+            $arguments.Add($source_password) > $null
+        }
+    }
+    if ($null -ne $certificate) {
+        $arguments.Add("--cert") > $null
+        $arguments.Add($certificate) > $null
+    }
+    if ($null -ne $certificate_password) {
+        $arguments.Add("--certpassword") > $null
+        if ($encrypted_certificate_password) {
+            # use a placeholder for now
+            $arguments.Add("pass") > $null
+        } else {
+            $arguments.Add($certificate_password) > $null
+        }
+    }
+    if ($null -ne $priority) {
+        $arguments.Add("--priority") > $null
+        $arguments.Add($priority) > $null
+    } else {
+        $priority = 0
+    }
+    if ($bypass_proxy -eq $true) {
+        $arguments.Add("--bypass-proxy") > $null
+    } else {
+        $bypass_proxy = $false
+    }
+    if ($allow_self_service -eq $true) {
+        $arguments.Add("--allow-self-service") > $null
+    } else {
+        $allow_self_service = $false
+    }
+    if ($admin_only -eq $true) {
+        $arguments.Add("--admin-only") > $null
+    } else {
+        $admin_only = $false
+    }
+
+    if ($check_mode) {
+        $arguments.Add("--what-if") > $null
+    }
+
+    $command = Argv-ToString -arguments $arguments
+    $res = Run-Command -command $command
+    if ($res.rc -ne 0) {
+        Fail-Json -obj $result -message "Failed to add Chocolatey source '$name': $($_.stderr)"
```

Comment:
I was surprised to see **$_** in this context, should that not be **$rc.stderr** instead ?

Reply:
You are right, joys of copying and pasting liberally.